tools/backup/backup.sh

```diff
@@ -74,6 +74,34 @@ function cloud_copy() {
     gsutil cp $backup_path $bucket_path
     gsutil cp $backup_path "${bucket_path}${LATEST_POINTER}"
     ;;
+  azure)
+    # Container is specified via BUCKET input, which can contain a path, i.e.
+    # my-container/foo
+    # AZ CLI doesn't allow this so we need to split it into container and container path.
+    CONTAINER=$(echo $BUCKET | sed 's/\/.*//')
+
+    # See: https://stackoverflow.com/a/10987027
+    CONTAINER_PATH=${BUCKET#$CONTAINER}
+    if [ "${CONTAINER_PATH: -1}" = "/" ]; then
+       CONTAINER_PATH=$(basename ${CONTAINER_PATH})
+    fi
+    CONTAINER_FILE=$CONTAINER_PATH/$database/$(basename "$backup_path")
```

Comment:
Did some local testing and the following two extra lines solves it:
```
CONTAINER_FILE=${CONTAINER_FILE#/}
CONTAINER_FILE=${CONTAINER_FILE#/}
```
And the output of my simple bash test with a hardcoded db/file:
```
➜  workdir ./test.sh dev/
Azure storage blob copy to dev :: db/file
➜  workdir ./test.sh dev
Azure storage blob copy to dev :: db/file
➜  workdir ./test.sh dev/test
Azure storage blob copy to dev :: test/db/file
➜  workdir ./test.sh dev/test/
Azure storage blob copy to dev :: test/db/file
```

And feel that all 4 cases should work and will see if I can find a way of removing 0,1 or 2 leading slashes in a better way.

Reply:
And found a way to cover it in one line and a comment suggestion:
```
# Remove all leading slashes to avoid creating empty folders in azure
CONTAINER_FILE=$(echo "$CONTAINER_FILE" | sed 's|^/*||')
```
And tested it so it works in all 4 cases.